spec/controllers/collection_participants_controller_spec.rb

```diff
@@ -0,0 +1,464 @@
+              it_redirects_to_with_notice(root_path, "You are now a member of #{collection.title}.")
+            end
+          end
+
+          context "where the user is not currently invited" do
+            it "redirects to the index and displays a notice that the user has already joined or applied" do
+              get :join, collection_id: collection.name
+              it_redirects_to_with_notice(root_path, "You have already joined (or applied to) this collection.")
+            end
+          end
+        end
+
+        context "where user isn't a participant yet" do
+          it "creates a participant for the user, redirects to index and notifies them that they have applied" do
+            get :join, collection_id: collection.name
+            participant = CollectionParticipant.find_by_pseud_id(user.default_pseud.id)
+            expect(participant).to be_present
+            expect(participant.participant_role).to eq CollectionParticipant::NONE
+            it_redirects_to_with_notice(root_path, "You have applied to join #{collection.title}.")
+          end
+        end
+      end
+    end
+  end
+
+  describe "index" do
+    let(:collection) { FactoryGirl.create(:collection) }
+    let(:current_role) { CollectionParticipant::NONE }
+    let(:user) { FactoryGirl.create(:user) }
+
+    context "user is not logged in" do
+      it "redirects to the index and displays an access denied message" do
+        get :index, collection_id: collection.name
+        it_redirects_to_with_error(new_user_session_path, "Sorry, you don't have permission to access the page you were trying to reach. Please log in.")
+      end
+    end
+
+    context "user is logged in" do
+      let!(:participant) do
+        FactoryGirl.create(
+          :collection_participant,
+          collection: collection,
+          pseud: user.default_pseud,
+          participant_role: current_role
+        )
+      end
+      let(:params) { { collection_id: collection.name } }
+
+      before do
+        fake_login_known_user(user)
+      end
+
+      context "where the user is not a maintainer" do
+        it "redirects to the index" do
+          get :index, params
+          it_redirects_to_with_error(user_path(user), "Sorry, you don't have permission to access the page you were trying to reach.")
+        end
+      end
+
+      context "where the user is a maintainer" do
+        render_views
+        let(:current_role) { CollectionParticipant::MODERATOR }
+
+        context "where the collection has several participants" do
+          let!(:users) do
+            Array.new(3) do
+              user = FactoryGirl.create(:user)
+              FactoryGirl.create(
+                :collection_participant,
+                collection: collection,
+                pseud: user.default_pseud
+              )
+              user
+            end
+          end
+
+          it "displays the participants in the correct order" do
```

Comment:
Does this actually test whether the users are in a specific order, or just that the users are present? I think it will pass even if they're higgledy-piggledy.